lib/rendezvous.rb

```diff
@@ -89,7 +89,13 @@ def socket
       host, port, secret = uri.host, uri.port, uri.path[1..-1]
 
       ssl_context = OpenSSL::SSL::SSLContext.new
-      ssl_context.ssl_version = :TLSv1
+
+      if ssl_context.respond_to?(:min_version=)
+        ssl_context.min_version = :TLS1
+        ssl_context.max_version = :TLS1_2
```

Comment:
Sorry I just noticed the symbols here don't have the "v" (ie they're `:TLS1` instead of `:TLSv1`). The [docs](https://docs.ruby-lang.org/en/2.5.0/OpenSSL/SSL/SSLContext.html#method-i-min_version-3D) don't really make it clear what the allowed symbols are (they use `:TLS1_2` in the example so I _assume_ that one is ok, but can't see `:TLS1` anywhere). You know, there are no tests, so I thought I'd better check 😀

Reply:
Yeah, it's super wacky, but `min_version` and `max_version` wants a different set of symbols than `ssl_version` does.

I still have my pry session around from when I was figuring it out:

```
[1] pry(main)> OpenSSL::SSL::SSLContext::METHODS
(pry):1: warning: constant OpenSSL::SSL::SSLContext::METHODS is deprecated
=> [:SSLv23, :SSLv23_client, :SSLv23_server, :SSLv2, :SSLv2_client, :SSLv2_server, :SSLv3, :SSLv3_client, :SSLv3_server, :TLSv1, :TLSv1_client, :TLSv1_server, :TLSv1_1, :TLSv1_1_client, :TLSv1_1_server, :TLSv1_2, :TLSv1_2_client, :TLSv1_2_server]
[2] pry(main)> OpenSSL::SSL::SSLContext
=> OpenSSL::SSL::SSLContext
[3] pry(main)> OpenSSL::SSL::SSLContext.new
=> #<OpenSSL::SSL::SSLContext:0x00007f897f308490>
[4] pry(main)> OpenSSL::SSL::SSLContext.new.ssl_version
NoMethodError: undefined method `ssl_version' for #<OpenSSL::SSL::SSLContext:0x00007f897f323290>
Did you mean?  ssl_version=
[5] pry(main)> OpenSSL::SSL::SSLContext.new.min_version
NoMethodError: undefined method `min_version' for #<OpenSSL::SSL::SSLContext:0x00007f897e74e230>
Did you mean?  min_version=
               max_version=
[6] pry(main)> OpenSSL::SSL::SSLContext.new.min_version = :TLS1_0
ArgumentError: unrecognized version "TLS1_0"
[7] pry(main)> OpenSSL::SSL::SSLContext.new.min_version = :TLSv1_0
ArgumentError: unrecognized version "TLSv1_0"
[8] pry(main)> OpenSSL::SSL::SSLContext.new.min_version = :TLSv_1_0_client
ArgumentError: unrecognized version "TLSv_1_0_client"
[9] pry(main)> OpenSSL::SSL::SSLContext.new.min_version = :TLSv_1_1_client
ArgumentError: unrecognized version "TLSv_1_1_client"
[10] pry(main)> OpenSSL::SSL::SSLContext.new.min_version = :TLSv1_1_client
ArgumentError: unrecognized version "TLSv1_1_client"
[11] pry(main)> OpenSSL::SSL::SSLContext.new.min_version = :TLSv1
ArgumentError: unrecognized version "TLSv1"
[12] pry(main)> OpenSSL::SSL::TLS1_1_VERSION
=> 770
[13] pry(main)> OpenSSL::SSL::SSLContext.new.min_version = :TLS1_1
=> :TLS1_1
[14] pry(main)> OpenSSL::SSL::SSLContext.new.min_version = :TLS1
=> :TLS1
[15] pry(main)> OpenSSL::SSL::SSLContext.new.min_version = :TLS1_0
ArgumentError: unrecognized version "TLS1_0"
[16] pry(main)> OpenSSL::SSL::SSLContext.new.min_version = :TLS1
=> :TLS1
[17] pry(main)> OpenSSL::SSL::SSLContext.new.min_version = :TLS1_1
=> :TLS1_1
[18] pry(main)> OpenSSL::SSL::SSLContext.new.min_version = :TLS1_2
=> :TLS1_2
```

You can provide constants as arguments instead, but I found that some of the constants weren't defined in older rubies, so I didn't want to lean on them.

Reply:
😄Nice, good sleuthing. Cool then, I can't see anything else (already gave my approval)!